test/test_xla_sharding.py

```diff
@@ -147,6 +147,27 @@ def test_metrics_recorded(self):
     self.assertIn("VirtualDeviceUsage", met.counter_names())
     self.assertNotEqual(met.counter_value("VirtualDeviceUsage"), 0)
 
+  def test_outbound_data_metrics(self):
+    baseline_outbound_data = met.counter_value("OutboundData")
+    partition_spec = (0, 1)
+    xt1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
+                       dtype=torch.float,
+                       device=xm.xla_device())
+    xs.mark_sharding(xt1, self._get_mesh((1, self.n_devices)), partition_spec)
+    outbound_with_virtual_device = met.counter_value(
+        "OutboundData") - baseline_outbound_data
+
+    os.environ["XLA_USE_SPMD"] = "0"
```

Comment:
do we need to restore this environment after this test? I think it is better to set it to 1, run `mark_sharding`, set it to 0, run again, compare the result and then restore the value of `XLA_USE_SPMD`

Reply:
Done

Reply:
I saw this test always set
```
os.environ["XLA_USE_SPMD"] = "1"
```
in the setup so this is fine.